src/types/utils.cpp

```diff
@@ -670,8 +672,15 @@ std::tuple<std::wstring, std::wstring> Utils::MangleStartingDirectoryForWSL(std:
         }
     } while (false);
 
+    // GH #12353: `~` is never a valid windows path. We can only accept that as
+    // a startingDirectory when the exe is specifically wsl.exe, because that
+    // can override the real startingDirectory. If the user set the
+    // startingDirectory to ~, but the commandline to something like pwsh.exe,
+    // that won't actually work. In that case, mangle the startingDirectory to
+    // %userprofile%, so it's at least something reasonable.
     return {
         std::wstring{ commandLine },
-        std::wstring{ startingDirectory }
+        std::wstring{ startingDirectory == L"~" ? wil::ExpandEnvironmentStringsW<std::wstring>(L"%USERPROFILE%") :

```

Comment:
can we just return a null startingdirectory and let the OS take care of it? or should we always resolve userprofile?

Reply:
We certainly could. I thought `userprofile` would be closer to what users expect, than... system32? Whatever the terminal's cwd is? `ubuntu.exe` ignores it's CWD, but I don't know about other launchers, so I dunno.... pwsh would use the startingDirectory I guess

Reply:
Okay so replacing the starting directory with `""` then gets treated as "Use Parent Process Directory". (which is basically the same as... `.`? )

IMO the userprofile one is more predictable, but overall I don't care one way or the other.

Reply:
I pushed that change to [`dev/migrie/b/12353-with-null`](https://github.com/microsoft/terminal/compare/dev/migrie/b/12353-with-null), if we make a decision and need it merged tonight.

Reply:
fine w userprofile. should we strip the unused variable?

Reply:
> fine w userprofile. should we strip the unused variable?

butts that was supposed to be pushed